hugr-core/src/import.rs

```diff
@@ -1263,6 +1291,116 @@ impl<'a> Context<'a> {
 
         Ok(json_value)
     }
+
+    fn import_value(
+        &mut self,
+        term_id: model::TermId,
+        type_id: model::TermId,
+    ) -> Result<Value, ImportError> {
+        let term_data = self.get_term(term_id)?;
+
+        match term_data {
+            model::Term::Wildcard => Err(error_uninferred!("wildcard")),
+            model::Term::Apply { .. } => {
+                Err(error_uninferred!("application with implicit parameters"))
+            }
+            model::Term::Var(_) => Err(error_unsupported!("constant value containing a variable")),
+
+            model::Term::ApplyFull { symbol, args } => {
+                let symbol_name = self.get_symbol_name(*symbol)?;
+
+                if symbol_name == TERM_JSON_CONST {
+                    let value = args.get(1).ok_or(model::ModelError::TypeError(term_id))?;
+
+                    let model::Term::Str(json) = self.get_term(*value)? else {
+                        return Err(model::ModelError::TypeError(term_id).into());
+                    };
+
+                    // We attempt to deserialize as the custom const directly.
+                    // This might fail due to the custom const struct not being included when
+                    // this code was compiled; in that case, we fall back to the serialized form.
+                    let value: Option<Box<dyn CustomConst>> = serde_json::from_str(json).ok();
+
+                    if let Some(value) = value {
+                        let opaque_value = OpaqueValue::from(value);
+                        return Ok(Value::Extension { e: opaque_value });
+                    } else {
+                        let runtime_type =
```

Comment:
I think we need test coverage of this case

Reply:
Test case added.